sycl/source/platform.cpp

```diff
@@ -95,6 +95,40 @@ context platform::ext_oneapi_get_default_context() const {
   return detail::createSyclObjFromImpl<context>(It->second);
 }
 
+std::vector<device> platform::ext_oneapi_get_composite_devices() const {
+  // Only some Intel GPU architectures can be composite devices.
+  auto GPUDevices = get_devices(info::device_type::gpu);
+  // Using ZE_FLAT_DEVICE_HIERARCHY=COMBINED, we receive tiles as devices, which
+  // are component devices. Thus, we need to get the composite device for each
+  // of the component devices, and filter out duplicates.
+  std::vector<device> Composites;
+  std::vector<device> Result;
+  for (auto &Dev : GPUDevices) {
+    if (Dev.has(sycl::aspect::ext_oneapi_is_component)) {
```

Comment:
early continue instead, please.

Reply:
Done in https://github.com/intel/llvm/pull/12178/commits/8d68661b1c4a6e35ea6ab80d0b5404071bbac958.